libkv/subscription.go

```diff
@@ -65,6 +65,48 @@ func (service Service) CreateSubscription(s *subscription.Subscription) (*subscr
 	return s, nil
 }
 
+// UpdateSubscription updates subscription.
+func (service Service) UpdateSubscription(id subscription.ID, s *subscription.Subscription) (*subscription.Subscription, error) {
+	err := service.validateSubscription(s)
```

Comment:
I know that you didn't introduce that but can you remove validateSubscription method from the struct and just make it regular function like `validateSubscriptionUpdate`?